File: TG1.1 SOAP_JAVA_EUREKABANK_GR03/01.SERVIDOR/WS_EurekaBank_SOAPJAVA_GR03/src/main/java/ec/edu/monster/model/DatosCuenta.java
package ec.edu.monster.model;

import jakarta.xml.bind.annotation.XmlElement;
import jakarta.xml.bind.annotation.XmlRootElement;
import java.math.BigDecimal;

/**
 *
 * @author leona
 */

@XmlRootElement(name = "GetDatosCuentaResult")
public class DatosCuenta {

    private String codigo;
    private String nombreCliente;
    private BigDecimal saldo;
    private String estado;
    private String moneda;
    private String emailCliente;
    private String telefonoCliente;

    // Constructor vacío (JAXB)
    public DatosCuenta() {
    }

    // Getters y Setters con @XmlElement para nombres personalizados
    @XmlElement(name = "Codigo")
    public String getCodigo() {
        return codigo;
    }

    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

    @XmlElement(name = "NombreCliente")
    public String getNombreCliente() {
        return nombreCliente;
    }

    public void setNombreCliente(String nombreCliente) {
        this.nombreCliente = nombreCliente;
    }

    @XmlElement(name = "Saldo")
    public BigDecimal getSaldo() {
        return saldo;
    }

    public void setSaldo(BigDecimal saldo) {
        this.saldo = saldo;
    }

    @XmlElement(name = "Estado")
    public String getEstado() {
        return estado;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    @XmlElement(name = "Moneda")
    public String getMoneda() {
        return moneda;
    }

    public void setMoneda(String moneda) {
        this.moneda = moneda;
    }

    @XmlElement(name = "EmailCliente")
    public String getEmailCliente() {
        return emailCliente;
    }

    public void setEmailCliente(String emailCliente) {
        this.emailCliente = emailCliente;
    }

    @XmlElement(name = "TelefonoCliente")
    public String getTelefonoCliente() {
        return telefonoCliente;
    }

    public void setTelefonoCliente(String telefonoCliente) {
        this.telefonoCliente = telefonoCliente;
    }
}

File: TG1.1 SOAP_JAVA_EUREKABANK_GR03/02.CLIMOV/CLIMOV_EurekaBank_SOAPJAVA_GR03/app/src/main/java/ec/edu/monster/models/DatosCuenta.kt

package ec.edu.monster.models

import org.ksoap2.serialization.KvmSerializable
import org.ksoap2.serialization.PropertyInfo
import java.math.BigDecimal
import java.util.Hashtable

data class DatosCuenta(
    var codigo: String = "",
    var nombreCliente: String = "",
    var saldo: BigDecimal = BigDecimal.ZERO,
    var estado: String = "",
    var moneda: String = "",
    var emailCliente: String = "",
    var telefonoCliente: String = ""
) : KvmSerializable {

    override fun getProperty(index: Int): Any = when (index) {
        0 -> codigo
        1 -> nombreCliente
        2 -> saldo
        3 -> estado
        4 -> moneda
        5 -> emailCliente
        6 -> telefonoCliente
        else -> throw IndexOutOfBoundsException()
    }

    override fun getPropertyCount(): Int = 7

    override fun setProperty(index: Int, value: Any) {
        when (index) {
            0 -> codigo = value.toString()
            1 -> nombreCliente = value.toString()
            2 -> saldo = if (value is BigDecimal) value else BigDecimal(value.toString())
            3 -> estado = value.toString()
            4 -> moneda = value.toString()
            5 -> emailCliente = value.toString()
            6 -> telefonoCliente = value.toString()
            else -> throw IndexOutOfBoundsException()
        }
    }

    override fun getPropertyInfo(index: Int, properties: Hashtable<*, *>?, info: PropertyInfo) {
        when (index) {
            0 -> { info.name = "Codigo"; info.type = PropertyInfo.STRING_CLASS }
            1 -> { info.name = "NombreCliente"; info.type = PropertyInfo.STRING_CLASS }
            2 -> { info.name = "Saldo"; info.type = BigDecimal::class.java }
            3 -> { info.name = "Estado"; info.type = PropertyInfo.STRING_CLASS }
            4 -> { info.name = "Moneda"; info.type = PropertyInfo.STRING_CLASS }
            5 -> { info.name = "EmailCliente"; info.type = PropertyInfo.STRING_CLASS }
            6 -> { info.name = "TelefonoCliente"; info.type = PropertyInfo.STRING_CLASS }
        }
    }
}